### PR TITLE
feat(openbdap): localizzazione section, explicit truncation notes, UTF-8 dumps

### PR DESCRIPTION
**Pull request by aborruso**
## openbdap

Il catalogo Open Data della Ragioneria Generale dello Stato da terminale: ricerca offline, righe filtrate per nome di colonna leggibile e opere pubbliche cercabili per CUP, CIG o codice fiscale.

**API:** openbdap | **Category:** other | **Press version:** 4.32.2
**Spec:** not specified in the manifest

### Publication Path

Reprint/replace of the merged `library/other/openbdap` entry (added in #2013). No other PR is open on this slug.

### What changed in this reprint

Fixes found while wiring this CLI into a public-works monitoring workflow, all measured against the live portal.

- **`dossier` gains a `localizzazione` section.** The MOP localizzazione dataset is national, so it is queried without a region filter, and its rows carry the six-digit ISTAT municipality code built from the separate province and municipality columns the portal publishes.
- **The region now comes from data instead of a name match.** `dossier` used to look for a region name inside `Descrizione Titolare`, which fails for most municipal holders (`COMUNE DI PALERMO` resolved to nothing) and fanned the search out over all 21 regional partitions. It now reads the ISTAT region code from the localizzazione rows and keeps the old heuristic as a declared fallback: a Palermo CUP went from 34.7s to 9.0s with `--no-cache`.
- **Truncation is no longer silent.** `righe`, `cup`, `cig` and the `dossier` sections say when a result stops exactly on `--limite`. For `righe` the note carries the true total from `conta` (50 of 347 rows on the measured case) and rides in `meta`, so `results` stays an array for existing consumers, plus a line on stderr for humans.
- **Half the requests of a CIG search were duplicate schema reads.** The 21 `gare` and 21 `partecipanti` datasets have identical column sets, identifiers included (verified on all 42), so `cercaNeiMOP` now reads the schema once per family and falls back to a dataset's own columns if a borrowed schema makes its row query fail. `cig` with `--no-cache` went from 23.3s to 18.4s, and from 9.4s to 4.9s with `--rate-limit 0`.
- **`scarica` returns UTF-8.** The portal serves the dump in latin-1 while the OData path of the same CLI returns clean UTF-8. The conversion is streamed and decides the encoding at the first non-ASCII byte, so an already-UTF-8 stream passes through untouched; `--raw` keeps the portal's original bytes. The converted file is byte-identical to `iconv -f ISO-8859-1`.
- **An empty archive is distinguishable from a genuine zero.** About half of all CUPs are not in MOP, so `trovati: 0` is a plausible answer nobody double-checks. `cup`, `cig`, `dossier` and `opere` now carry `archivio_vuoto: true` beside the existing note. `--home` also moves the archive now, because its path is resolved at use instead of at command construction, which made cold tests read the real archive without saying so.
- **`sync` paginated only the first page.** The generated `resourceSupportsPagination` and `determinePaginationDefaults` switches were both empty, so `catalogo-package-search` fetched one page and reported `success` with `errored: 0`: 20 records where the endpoint returns 43 for `q=comuni`. Both switches now declare the endpoint's `start`/`rows` pagination.

Every code-level change is recorded under `.printing-press-patches/`.

The stale `publish-live-gate.json` proof is dropped rather than refreshed: it carries raw live API output and absolute paths from the generating host, and the phase5 acceptance marker beside it is the one the gates read.

### CLI Shape

```bash
$ openbdap-pp-cli --help
Openbdap CLI — Il catalogo Open Data della Ragioneria Generale dello Stato da terminale: ricerca offline, righe filtrate per nome di colonna leggibile e opere pubbliche cercabili per CUP, CIG o codice fiscale.

Highlights (not in the official API docs):
  • dossier   Il quadro completo di un'opera pubblica a partire dal CUP: progetto, pagamenti, gare, partecipanti, piano dei costi e soggetti titolari.
  • serie   Le annualita', le mensilita' e le regioni disponibili di una stessa serie di dataset.
  • opere   Le opere pubbliche in capo a un ente, a partire dal suo codice fiscale, con il totale reale.
  • novita   I dataset il cui ultimo aggiornamento cade nella finestra indicata.
  • campi   In quali dataset esiste un campo, con l'identificativo pronto da usare nei filtri.
  • mop   Quale dataset MOP interrogare per ogni regione e ruolo, con il relativo identificativo OData.
  • cig   La gara e i partecipanti a partire dal codice CIG.

Agent mode: add --agent to any command for JSON output + non-interactive mode.
Health check: run 'openbdap-pp-cli doctor' to verify connectivity.
See README.md or the bundled SKILL.md for recipes.

Usage:
  openbdap-pp-cli [command]

Available Commands:
  agent-context  Emit structured JSON describing this CLI for agents
  allinea        Allinea il catalogo dei dataset nell'archivio locale
  analytics      Run analytics queries on locally synced data
  api            Browse all API endpoints by interface name
  campi          In quali dataset esiste un campo, con l'identificativo per i filtri
  catalogo       Dataset del catalogo OpenBDAP
  cerca          Cerca dataset nell'archivio locale, per titolo e descrizione
  cig            Cerca una gara e i suoi partecipanti dal codice CIG
  colonne        Mostra le colonne di un dataset, con il nome da usare nei filtri
  conta          Conta le righe di un dataset, anche con un filtro
  cup            Cerca un progetto di opera pubblica dal suo codice CUP
  dati           Dati tabellari dei dataset via OData
  doctor         Check CLI health
  dossier        Il quadro completo di un'opera pubblica a partire dal CUP
  export         Export data to JSONL or JSON for backup, migration, or analysis
  gruppi         Temi (gruppi) del catalogo
  licenze        Elenca le licenze
  mop            Quale dataset MOP interrogare per ogni regione e ruolo
  novita         I dataset aggiornati di recente nell'archivio locale
  opere          Le opere pubbliche in capo a un ente, dal suo codice fiscale
  righe          Estrae le righe di un dataset, filtrando per nome di colonna leggibile
  scarica        Scarica l'intero dataset in CSV (separatore punto e virgola)
  search         Full-text search across synced data or live API
  serie          Le annualita', le mensilita' e le regioni disponibili di una serie
  sync           Sync API data to local SQLite for offline search and analysis
  tag            Elenca le parole chiave
  which          Find the command that implements a capability
  workflow       Compound workflows that combine multiple API operations
```

(The listing above is trimmed to the API-facing commands; the shipped binary also carries the standard printed-CLI commands: `completion`, `feedback`, `help`, `learnings`, `playbook`, `profile`, `recall`, `tail`, `teach`, `teach-lookup`, `teach-pattern`, `teach-playbook`, `version`.)

### Novel Commands

| Command | Name | Description |
|---------|------|-------------|
| `dossier` | Dossier di progetto | Il quadro completo di un'opera pubblica a partire dal CUP: progetto, pagamenti, gare, partecipanti, piano dei costi e soggetti titolari. |
| `serie` | Serie storiche del catalogo | Le annualita', le mensilita' e le regioni disponibili di una stessa serie di dataset. |
| `opere` | Opere di un ente | Le opere pubbliche in capo a un ente, a partire dal suo codice fiscale, con il totale reale. |
| `novita` | Novita' del catalogo | I dataset il cui ultimo aggiornamento cade nella finestra indicata. |
| `campi` | Indice dei campi | In quali dataset esiste un campo, con l'identificativo pronto da usare nei filtri. |
| `mop` | Mappa della famiglia MOP | Quale dataset MOP interrogare per ogni regione e ruolo, con il relativo identificativo OData. |
| `cig` | Ricerca per CIG | La gara e i partecipanti a partire dal codice CIG. |

### What This CLI Does

OpenBDAP pubblica migliaia di dataset di finanza pubblica dietro un'API CKAN con conteggi sbagliati e un OData con nomi di colonna offuscati e senza elenco. Questa CLI allinea il catalogo in un archivio SQLite locale con `allinea`, ci costruisce sopra una ricerca full-text su titolo e descrizione, e traduce i nomi leggibili delle colonne negli identificativi che il servizio pretende. Comandi come `dossier`, `opere` e `serie` fanno in un colpo cio' che oggi richiede una manciata di chiamate costruite a mano.

### Manuscripts

- [Research Brief](https://github.com/mvanhorn/printing-press-library/blob/8c1c0f0d791c6bf7b9f3ef293291bebdb34f4158/library/other/openbdap/.manuscripts/20260917-200055-21d6873c/research/2026-09-17-201258-feat-openbdap-pp-cli-brief.md)
- [Absorb Manifest](https://github.com/mvanhorn/printing-press-library/blob/8c1c0f0d791c6bf7b9f3ef293291bebdb34f4158/library/other/openbdap/.manuscripts/20260917-200055-21d6873c/research/2026-09-17-203113-feat-openbdap-pp-cli-absorb-manifest.md)
- [Novel Features Brainstorm](https://github.com/mvanhorn/printing-press-library/blob/8c1c0f0d791c6bf7b9f3ef293291bebdb34f4158/library/other/openbdap/.manuscripts/20260917-200055-21d6873c/research/2026-09-17-203113-novel-features-brainstorm.md)
- [Archived Spec](https://github.com/mvanhorn/printing-press-library/blob/8c1c0f0d791c6bf7b9f3ef293291bebdb34f4158/library/other/openbdap/.manuscripts/20260917-200055-21d6873c/research/openbdap-spec.yaml)
- [Build Log](https://github.com/mvanhorn/printing-press-library/blob/8c1c0f0d791c6bf7b9f3ef293291bebdb34f4158/library/other/openbdap/.manuscripts/20260917-200055-21d6873c/proofs/2026-09-17-205432-fix-openbdap-pp-cli-build-log.md)
- [Shipcheck Log](https://github.com/mvanhorn/printing-press-library/blob/8c1c0f0d791c6bf7b9f3ef293291bebdb34f4158/library/other/openbdap/.manuscripts/20260917-200055-21d6873c/proofs/2026-09-17-213632-fix-openbdap-pp-cli-shipcheck.md)
- [Dogfood Log](https://github.com/mvanhorn/printing-press-library/blob/8c1c0f0d791c6bf7b9f3ef293291bebdb34f4158/library/other/openbdap/.manuscripts/20260917-200055-21d6873c/proofs/2026-09-17-224921-fix-openbdap-pp-cli-dogfood.md)
- [Phase 4.85 Findings](https://github.com/mvanhorn/printing-press-library/blob/8c1c0f0d791c6bf7b9f3ef293291bebdb34f4158/library/other/openbdap/.manuscripts/20260917-200055-21d6873c/proofs/phase-4.85-findings.md)
- [Phase 4.95 Findings](https://github.com/mvanhorn/printing-press-library/blob/8c1c0f0d791c6bf7b9f3ef293291bebdb34f4158/library/other/openbdap/.manuscripts/20260917-200055-21d6873c/proofs/phase-4.95-findings.md)
- [Phase 5 Acceptance](https://github.com/mvanhorn/printing-press-library/blob/8c1c0f0d791c6bf7b9f3ef293291bebdb34f4158/library/other/openbdap/.manuscripts/20260917-200055-21d6873c/proofs/phase5-acceptance.json)

### Validation Results

| Check | Result |
|-------|--------|
| Manifest | PASS |
| Phase 5 | PASS |
| go mod tidy | PASS |
| govulncheck (this CLI only, reachable findings) | PASS |
| go vet | PASS |
| go build | PASS |
| --help | PASS |
| --version | PASS |
| verify-skill | PASS |
| patches | PASS |
| Manuscripts | PRESENT |

`go test ./...` passes in the source tree, including the new unit tests for the ISTAT code concatenation, the region resolver, the truncation detector and the latin-1 reader.

### Publish Live Gate

Full live dogfood reran at publish time against the real portal and passed: 246 tests, 0 failures. Proof: [Phase 5 Acceptance](https://github.com/mvanhorn/printing-press-library/blob/8c1c0f0d791c6bf7b9f3ef293291bebdb34f4158/library/other/openbdap/.manuscripts/20260917-200055-21d6873c/proofs/phase5-acceptance.json).

### Gaps

- The manifest carries no `spec_url`: the CLI was printed from a locally archived spec, which ships under `.manuscripts/<run-id>/research/openbdap-spec.yaml`.
- The `dossier` description in `.printing-press.json`, and therefore in the README and SKILL feature lists, still predates the `localizzazione` section. It was left untouched so the documented description keeps matching the generated command surfaces; the section is documented in the README troubleshooting list and in the command's own long help.

### CODE REVIEW

File: library/other/openbdap/internal/cli/openbdap_scarica.go [hunk 1]
```diff
@@ -155,5 +168,107 @@ func newScaricaCmd(flags *rootFlags) *cobra.Command {
...
+	prefisso, _ := l.src.Peek(l.src.Size())
+	for i, b := range prefisso {
+		if b < 0x80 {
+			continue
+		}
+		fine := i + 4
+		if fine > len(prefisso) {
+			fine = len(prefisso)
+		}
+		if r, _ := utf8.DecodeRune(prefisso[i:fine]); r == utf8.RuneError {
```
[the remark this answers is not in the document]
Comment (aborruso) [reply library/other/openbdap/internal/cli/openbdap_scarica.go]: Right on both counts, fixed in 9c50136.

The boundary case was a real corruption path: with the first accented character in the last bytes of the 64 KiB peek, `DecodeRune` saw a cut sequence, called the stream latin-1 and re-encoded an already-UTF-8 dump. The added test measures it on the pre-fix code as 65542 bytes where the input had 65540. A sequence cut by the buffer boundary is not an invalid sequence, so it no longer decides anything: the reader consumes only the ASCII bytes that precede it and looks again once `bufio` has refilled, which is safe because on ASCII the conversion is the identity. At end of stream a truncated sequence really is truncated, and latin-1 stays the answer.

The ambiguous prefix is handled by widening what the decision looks at: `utf8.Valid` over the whole peeked block instead of the first high byte. A single byte pair can form a valid sequence by accident, as `\xc3\xa8` does; a whole latin-1 CSV of Italian place names practically always carries a sequence that does not stand up as UTF-8. Three tests cover the three shapes: the boundary case, a latin-1 block whose first high pair is valid UTF-8 by accident, and a truncated tail at EOF.

You are right that no prefix is a proof, only evidence. Two things bound the residual risk here: the endpoint is one portal's CSV dump, served today with `Content-Type: text/csv` and latin-1 bytes, and a declared `charset=utf-8` short-circuits the detection before it runs. For the case where evidence is not enough there is `--raw`, which returns the portal's bytes untouched.

**Comment by aborruso** [comment]
**Comments Outside Diff — Archive path is misreported (P2), `openbdap_comandi_base.go:160`**

Valid, fixed in 9c50136.

`allinea` now resolves the archive path once, at the top of the run, and uses that one value for the store, for the `archivio` field of the JSON envelope and for the human line. Before, only the store open went through the resolver, so with `--home` and no explicit `--db` the command wrote to the relocated archive and reported the path computed when the command was constructed. That is the same class of defect as the `--home` fix itself: a command that is right about what it does and wrong about what it says.
